src/viam/trajex/totg/trajectory.cpp

```diff
@@ -360,48 +369,50 @@ enum class integration_event : std::uint8_t {
                         const auto after_extremum = std::min(*first_extremum + arc_length{opt.epsilon}, current_segment.end());
 
                         // Skip validation if extremum is too close to segment boundaries (can't get clean epsilon separation)
-                        const bool too_close_to_start = (*first_extremum - before_extremum) < (opt.epsilon * 0.5);
-                        const bool too_close_to_end = (after_extremum - *first_extremum) < (opt.epsilon * 0.5);
+                        const auto half_epsilon = opt.epsilon * 0.5;
+                        const auto too_close_to_start = (half_epsilon.wrap(*first_extremum) == half_epsilon.wrap(before_extremum));
+                        const bool too_close_to_end = (half_epsilon.wrap(after_extremum) == half_epsilon.wrap(*first_extremum));
 
                         if (too_close_to_start || too_close_to_end) {
                             first_extremum.reset();
+                            return;
+                        }
+
+                        const auto q_prime_before = current_segment.tangent(before_extremum);
+                        const auto q_double_prime_before = current_segment.curvature(before_extremum);
+                        const auto [s_dot_before, _1] = compute_velocity_limits(
+                            q_prime_before, q_double_prime_before, opt.max_velocity, opt.max_acceleration, opt.epsilon);
+
+                        const auto q_prime_after = current_segment.tangent(after_extremum);
+                        const auto q_double_prime_after = current_segment.curvature(after_extremum);
+                        const auto [s_dot_after, _2] = compute_velocity_limits(
+                            q_prime_after, q_double_prime_after, opt.max_velocity, opt.max_acceleration, opt.epsilon);
+
+                        // Check if curve has local minimum (decreasing before, increasing after)
+                        const bool is_local_minimum = (s_dot_before > s_dot_max_acc) && (s_dot_after > s_dot_max_acc);
+
+                        if (is_local_minimum) {
+                            // Valid continuous-nondifferentiable switching point (Section VII-A case 2).
+                            // Backward integration starts from this point with zero acceleration, since applying
+                            // any non-zero acceleration would move off the non-differentiable cusp. The switching
+                            // velocity is the value of the limit curve at this point.
+                            first_extremum_velocity = s_dot_max_acc;
```

Comment:
This just got un-indented because now we have the early return above rather than an else.